src/shared/generator.c

```diff
@@ -620,6 +621,11 @@ int generator_write_cryptsetup_service_section(
 }
 
 void log_setup_generator(void) {
-        log_set_prohibit_ipc(true);
-        log_setup_service();
+        /* Disable talking to syslog/journal (i.e. the two IPC-based loggers) if we run in system context. */
+        if (cg_pid_get_owner_uid(0, NULL) == -ENXIO /* not running in a per-user slice */)
+                log_set_prohibit_ipc(true);
+
```

Comment:
yes. journald is socket activated. if we allow generators to log to its sockets things might deadlock since PID 1 might wait for us to finish and thus will not handle socket activation.

After all oyu never know when "systemctl daemon-reload" is issued: could be before journald is started or after, you never know. Hence the only safe thing is to systematically say: no IPC logging in generators.

(In the long run PID 1 really should start collecting stdout/stderr of generators and propagate them to its own log destination, as reqeuested in that other RFE issue, but as long as we don't do that, kmsg is the only place for system generators. And that's even documented in the generator man page)